src/orca/Deadline.java

```diff
@@ -0,0 +1,30 @@
+package orca;
+
+public class Deadline extends Task {
+    protected String by;
+
+    Deadline(String userInput, int startIdx) throws OrcaException {
+        try {
+            String[] deadline = userInput.substring(startIdx).split(" /by ");
+            this.description = deadline[0];
+            this.by = deadline[1];
```

Comment:
Do take note of magic numbers as there are quite a number of them appearing in the rest of your code! You can try to use a named constant makes the code easier to understand because the name tells us more about the meaning of the number.